src/scenario/scenario.cpp

```diff
@@ -528,3 +528,14 @@ cItem cScenario::return_treasure(int loot, bool allow_junk) {
 		treas.value = 0;
 	return treas;
 }
+
+cOutdoors& cScenario::get_sector(int x, int y)
+{
+	return *outdoors[x][y];
+}
+
+bool cScenario::town_exists(spec_loc_t loc) const
```

Comment:
Given that it takes a `spec_loc_t`, this is more "does the town entrance lead to a valid town" than "does this town exist" (which would just take the town index)...

I'm thinking maybe `city_valid` or something similar? Still not sure of the best name here.

This function could also be called with invalid data, since `spec_loc_t` is also used for things that have nothing to do with towns, but there's not a lot to be done about that (except maybe cloning the type for towns).

Reply:
will revise, i have an idea